src/app/modules/payments/stripe.utils.ts

```diff
@@ -54,25 +93,42 @@ const isStripeMetadata = (
  * it (i.e. payment intents and charges).
  * @param {Stripe.Metadata} metadata the metadata object which is expected to have a payment id
  * @returns ok(paymentId) the extracted paymentId
+ * @returns err(StripeMetadataInvalidError) if the metadata has an invalid shape
  * @returns err(StripeMetadataValidPaymentIdNotFoundError) if the payment id was not found or is an invalid BSON object id
+ * @returns err(StripeMetadataIncorrectEnvError) if the app is incorrect
  */
 export const getMetadataPaymentId = (
   metadata: Stripe.Metadata,
-): Result<string, StripeMetadataValidPaymentIdNotFoundError> => {
+): Result<
+  string,
+  | StripeMetadataInvalidError
+  | StripeMetadataValidPaymentIdNotFoundError
+  | StripeMetadataIncorrectEnvError
```

Comment:
hm would it make more sense to check for the env in a separate function instead of `getMetadataPaymentId`? then we can also do the check earlier and avoid processing the webhook (in `stripe.controller.ts`) if it's from the wrong env

Reply:
hmmm, we could but it would still be quite messy. The target of the call to `getMetadataPaymentId` is different based on different cases (see the `switch` in `handleStripeEventUpdates`), since the metadata to be checked exists in different places in each type of event. If in the future we want to extract other data from the metadata, we can expand the scope of this function to extract all relevant values. It's just that currently, we only need the payment id.

Reply:
ok gotcha! so it's mainly in `getMetadataPaymentId` because that's where we expect to process the metadata (?).